include/extractor/edge_based_graph_factory.hpp

```diff
@@ -42,6 +42,33 @@ namespace osrm
 namespace extractor
 {
 
+namespace lookup
+{
+// Set to 1 byte alignment
+#pragma pack(push, r1, 1)
```

Comment:
This is a trick to avoid wasted bytes in the files on disk.

https://en.wikipedia.org/wiki/Data_structure_alignment#Default_packing_and_.23pragma_pack

`#pragma pack` allow you to set the structure word alignment, and here I'm setting it to 1 byte, so there will be no padding.

The downside to doing this is that operations on structs that are not word aligned in memory could mean extra memory access and some calculation to join un-aligned data.

In this particular case, that overhead is dwarfed by the I/O we're doing, and for the data structures that aren't properly aligned, saves us a bunch of bytes on disk.  `SegmentBlock` is the only one that gets some padding in the default word alignment on x86-64 (4 extra bytes).

I'll add a clearer explanation as to why this is done.